### ISSUE TITLE
Hardcoded cluster label in victoria-metrics-k8s-stack

### ISSUE THREAD

**Issue by rlex** [VictoriaMetrics/helm-charts#774]
Hello, it seems like "cluster" label is [hardcoded in chart](https://github.com/VictoriaMetrics/helm-charts/blob/master/charts/victoria-metrics-k8s-stack/values.yaml#L585)

This can cause issues when label cluster is used elsewhere in cluster (redis cluster, mongo cluster, etc etc etc), but if it's specified in externalLabels, then it will get overwritten anyway.

i also seemingly can't remove it with setting values, because there is [merge happening](https://github.com/VictoriaMetrics/helm-charts/blob/master/charts/victoria-metrics-k8s-stack/templates/_helpers.tpl#L176-L178) and it gets added anyway, picking it from default values.

I guess this is used for multi-cluster setup, where multiple k8s clusters are writing to same VM?

Maybe make it more unique to not overlap with existing metrics? Like k8s_cluster, for example, or just give ability to drop it from default values.

**Comment by Haleygo** [comment]
Hello!
>Maybe make it more unique to not overlap with existing metrics? Like k8s_cluster, for example, or just give ability to drop it from default values.

As it's defined in values.yaml, you can always override the default value by [using your custom values.yaml](https://helm.sh/docs/chart_template_guide/values_files/) like `helm install [RELEASE_NAME] vm/victoria-metrics-k8s-stack -f customValues.yaml` 
or using `--set` when install like `helm install [RELEASE_NAME] vm/victoria-metrics-k8s-stack --set vmagent.spec.externalLabels={}`

**Comment by rlex** [comment]
> Hello!
> 
> > Maybe make it more unique to not overlap with existing metrics? Like k8s_cluster, for example, or just give ability to drop it from default values.
> 
> As it's defined in values.yaml, you can always override the default value by [using your custom values.yaml](https://helm.sh/docs/chart_template_guide/values_files/) like `helm install [RELEASE_NAME] vm/victoria-metrics-k8s-stack -f customValues.yaml` or using `--set` when install like `helm install [RELEASE_NAME] vm/victoria-metrics-k8s-stack --set vmagent.spec.externalLabels={}`

Sure i can do this that way, emptying externalLabels, but in that case i won't be able to use externalLabels at all.

**Comment by Haleygo** [comment]
>emptying externalLabels, but in that case i won't be able to use externalLabels at all.

Yeah, you can't remove the original `cluster` label with `--set`, only can change it's value like `--set vmagent.spec.externalLabels.<label-name>=<label-key>`.
And If you want to remove it, you need to [use your custom values.yaml](https://helm.sh/docs/chart_template_guide/values_files/).

>This can cause issues when label cluster is used elsewhere in cluster (redis cluster, mongo cluster, etc etc etc), but if it's specified in externalLabels, then it will get overwritten anyway.

I think maybe you should add [relabel](https://new.docs.victoriametrics.com/relabeling/) config for those scrape jobs, change them to something like `redis_cluster`, `mongo_cluster`, since `cluster` is pretty useful and exclusive in multi-cluster.

**Comment by SQLJames** [comment]
Good afternoon, 
I also ran into this on a demo for cloudnative-pg monitoring. When running 
`helm template vmstack vm/victoria-metrics-k8s-stack -f ./values.yaml > out.yaml`
with the following values
```
vmagent:
  spec:
    externalLabels: {}
```

the vmagent spec still contains the cluster label. 
```
...
# Source: victoria-metrics-k8s-stack/templates/victoria-metrics-operator/vmagent/vmagent.yaml
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMAgent
metadata:
  name: vmstack-victoria-metrics-k8s-stack
  namespace: default
  labels: 
    helm.sh/chart: victoria-metrics-k8s-stack-0.23.1
    app.kubernetes.io/name: victoria-metrics-k8s-stack
    app.kubernetes.io/instance: vmstack
    app.kubernetes.io/version: "v1.101.0"
    app.kubernetes.io/managed-by: Helm
spec:
  externalLabels:
    cluster: cluster-name
...
```
doing the same via the command line with `helm template vmstack vm/victoria-metrics-k8s-stack --set vmagent.spec.externalLabels={} > out.yaml` I get a warning stating 
```
coalesce.go:286: warning: cannot overwrite table with non table for victoria-metrics-k8s-stack.vmagent.spec.externalLabels (map[cluster:cluster-name])
``` 
I do believe that this is a flaw in the chart and should be changed especially since the externallabels field for the VMAlert yaml is set to an empty map
```
kind: VMAlert
metadata:
  name: vmstack-victoria-metrics-k8s-stack
  namespace: default
  labels: 
    helm.sh/chart: victoria-metrics-k8s-stack-0.23.1
    app.kubernetes.io/name: victoria-metrics-k8s-stack
    app.kubernetes.io/instance: vmstack
    app.kubernetes.io/version: "v1.101.0"
    app.kubernetes.io/managed-by: Helm
spec:
  datasource:
    url: http://vmsingle-vmstack-victoria-metrics-k8s-stack.default.svc:8429/
  evaluationInterval: 15s
  externalLabels: {}
```

**Comment by SQLJames** [comment]
@rlex were you able to get the scrape relabel done? I am fairly new and am struggling a bit getting it implemented with the stack chart, as I am not fully sure where it lives at the moment Haha.

**Comment by SQLJames** [comment]
Posting my last update for my work around , I ended up using an umbrella chart in helm and pointing to a local nested version of the stack chart, this allowed me to remove the externallabel for `cluster` and instead I replaced it with `victoriametrics/cluster="homelab1"` This allowed me to namespace the label and seemed to work well on the collection of the data as this is how a datapoint showed up. 
```
{__name__="cnpg_collector_up", cluster="awx", container="postgres", endpoint="metrics", instance="10.233.120.94:9187", job="postgres/awx", namespace="postgres", pod="awx-2", prometheus="monitoring/victoriametrics-vmstack", victoriametrics/cluster="homelab1"}
```

**Comment by rlex** [comment]
@SQLJames nope, i just switched to deploying operator only and every other resource as separate CR. Way more flexible this way.

**Comment by SQLJames** [comment]
Understood, thank you! I have a work around for the time being but I am not a fan of hosting the chart so switching to that methodology might be the better solution.

**Comment by Haleygo** [comment]
Hi @SQLJames @rlex ,
JFYI, the default `cluster` external label has been removed in https://github.com/VictoriaMetrics/helm-charts/pull/1062 and released in [victoria-metrics-k8s-stack-0.23.2](https://github.com/VictoriaMetrics/helm-charts/releases/tag/victoria-metrics-k8s-stack-0.23.2).
